launch.go

```diff
@@ -72,6 +73,31 @@ func init() {
 	client = retryablehttp.NewClient()
 }
 
+/* has HTTP or HTTPS protocol
+targetURL => URL
+*/
+func hasHTTPProtocol(targetURL *url.URL) bool {
+	return targetURL.Scheme == "http" || targetURL.Scheme == "https"
+}
+
+/* make pushgateway url
+baseURL => base url for pushgateway
+buildID => sd build id
+*/
+func makePushgatewayURL(baseURL string, buildID int) (string, error) {
+	var pushgatewayURL string = baseURL
+	u, err := url.Parse(pushgatewayURL)
+	if err != nil {
+		log.Printf("makePushgatewayURL: failed to parse url [%v], buildId:[%v], error:[%v]", pushgatewayURL, buildID, err)
+		return "", err
+	}
+	u.Path = path.Join(u.Path, "/metrics/job/containerd/instance/"+strconv.Itoa(buildID))
+	if !hasHTTPProtocol(u) {
+		return "http://" + u.String(), nil
```

Comment:
How about this?
```suggestion
		u.Scheme = "http"
```

Reply:
Can you add a test that adds a schema when an environment variable with no schema is specified?

Reply:
If pushgateway url has port and no protocol, `url.Parse` behaves strangely. So we can't use `u.Scheme` in this case.
Trying code is below. I fixed it for this case.
```
package main
import (
	"fmt"
	"log"
	"net/url"
)
func main() {
	u, err := url.Parse("pushgateway.monitoring:9091")
	if err != nil {
		log.Fatal(err)
	}
	if u.Scheme == "" {
		u.Scheme = "http"
	}
	fmt.Println(u.Scheme)  // => pushgateway.monitoring
	u.Path = u.Path + "/metrics/job/containerd/instance/"
	fmt.Println(u.String()) // => pushgateway.monitoring:9091
}
```

Reply:
> Can you add a test that adds a schema when an environment variable with no schema is specified?

I added a test when `SD_BUSHGATEWAY_URL` has no scheme.